client/app/pages/queries/hooks/useAutoLimitFlags.js

```diff
@@ -0,0 +1,24 @@
+import { useCallback, useMemo, useState } from "react";
+import localOptions from "@/lib/localOptions";
+import { get, extend } from "lodash";
+
+function isAutoLimitAvailable(dataSource) {
+  return get(dataSource, "apply_auto_limit", false);
+}
+
+export default function useAutoLimitFlags(dataSource, query, setQuery) {
+  const isAvailable = isAutoLimitAvailable(dataSource);
+  const [isChecked, setIsChecked] = useState(localOptions.get("ApplyAutoLimit", true));
+  query.options.applyAutoLimit = isAvailable && isChecked;
+
+  const setAutoLimit = useCallback(
+    state => {
+      setIsChecked(state);
+      localOptions.set("ApplyAutoLimit", state);
+      setQuery(extend(query.clone(), { options: { ...query.options, applyAutoLimit: isAvailable && state } }));
+    },
+    [query, setQuery, isAvailable]
+  );
+
+  return useMemo(() => [isAvailable, isChecked, setAutoLimit], [isAvailable, isChecked, setAutoLimit]);
```

Comment:
Why the `useMemo` here?

Reply:
It was probably just copied from other hooks 😅, but it is indeed safe to remove @gjxdxh